frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -1155,37 +1300,94 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
-                            query: val ?? '',
-                        },
-                    })
-                } else {
-                    actions.setSourceQuery({
-                        kind: NodeKind.DataVisualizationNode,
-                        source: {
-                            kind: NodeKind.HogQLQuery,
-                            query: val ?? '',
-                        },
-                        display: ChartDisplayType.ActionsLineGraph,
+        activeModelUri: async (activeModelUri) => {
+            if (!activeModelUri || !props.monaco) {
+                return
+            }
+
+            const _model = props.monaco.editor.getModel(activeModelUri.uri)
+            const val = _model?.getValue()
+            actions.setQueryInput(val ?? '')
+
+            const path = activeModelUri.uri.path.split('/').pop()
+            let storedVariables = {}
+            if (path) {
+                const storedVariablesString = await getStorageItem(`${activeModelVariablesStateKey(props.key)}_${path}`)
+                if (storedVariablesString) {
+                    try {
+                        storedVariables = JSON.parse(storedVariablesString)
+                    } catch (error) {
+                        posthog.captureException(new Error('sql-editor-variables-parse-failure'), { error, key })
+                    }
+                }
+            }
+
+            // Fix stored variables to ensure they have proper UUIDs
+            const fixedStoredVariables: Record<string, any> = {}
+            Object.entries(storedVariables).forEach(([key, variable]) => {
+                if (variable && typeof variable === 'object') {
+                    const varObj = variable as any
+                    if (
+                        !varObj.variableId ||
+                        typeof varObj.variableId !== 'string' ||
+                        varObj.variableId.length !== 36
+                    ) {
+                        fixedStoredVariables[key] = { ...varObj, variableId: crypto.randomUUID() }
+                    } else {
+                        fixedStoredVariables[key] = varObj
+                    }
+                }
+            })
+
+            // Extract variables from query if they exist
+            if (val && val.includes('{variables.')) {
+                const variableMatches = val.match(/\{variables\.([^}]+)\}/g)
+                if (variableMatches) {
+                    variableMatches.forEach((match) => {
+                        const codeName = match.replace('{variables.', '').replace('}', '')
+                        if (!fixedStoredVariables[codeName]) {
+                            fixedStoredVariables[codeName] = {
+                                variableId: crypto.randomUUID(),
+                                code_name: codeName,
+                                value: null,
+                                isNull: false,
+                            }
+                        }
```

Comment:
Can you leave a note on why we're doing all of this variable logic? What is this solving? I'm not aware of variables being an issue with what this PR was trying to solve